packages/synapse-react-client/src/testutils/ReactQueryMockUtils.ts

```diff
@@ -119,5 +117,5 @@ export function getUseMutationMock<TData, TError, TVariables>(
     failureReason: null,
     isPending: false,
     submittedAt: 0,
-  }
+  } satisfies UseMutationResult<TData, TError, TVariables>
```

Comment:
Using `satisfies` here gives us the type-checking we want while preserving the actual type.

The utility of this is that callers will now see that `mutate`/`mutateAsync`/`reset` properties are Jest mock functions, and can be interacted with as such without a type cast